### PR TITLE
Push to dockerhub via circleci

### PR DESCRIPTION
**Pull request by bqbn**
The v2 deployment pipeline leverages CircleCI to push images to Dockerhub. Thus it is necessary to add the docker image related build and push steps to the config file to make it happen.

### CODE REVIEW

**Comment by bqbn** [comment]
@EnTeQuAk or @davehunt r?

I'll do the merge and work out any kinks that may happen.

File: circle.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ machine:
 
 dependencies:
   override:
+    - docker version
+    - docker build -t app:build -f Dockerfile.deploy .
```
Comment (EnTeQuAk) [inline circle.yml]: lol, I didn't know about `-t` option for `build` and always wrote some wrapper scripts that got the image-id from the output. I like it 😆

File: circle.yml [hunk 2]
```diff
@@ -37,3 +39,26 @@ test:
...
+  releases:
+    # push all tags
+    tag: /.*/
+    commands:
+      - docker tag app:build ${DOCKERHUB_REPO}:${CIRCLE_TAG}
+      - docker login -e $DOCKERHUB_EMAIL -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
+      - docker push ${DOCKERHUB_REPO}:${CIRCLE_TAG}
+
+  branches:
+    branch: push-to-dockerhub-via-circleci
```
Comment (EnTeQuAk) [inline circle.yml]: What's this all about? Are you planning this for testing and delete these lines later?

**Comment by EnTeQuAk** [comment]
Before merging, please make sure to squash your commits :)

**Pull request merged**